pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java

```diff
@@ -91,92 +155,162 @@ public void run() {
     }
   }
 
-  public static void singleThreadedQueryRunner(PerfBenchmarkDriverConf conf, String queryFile)
+  /**
+   * Use multiple threads to run query at an increasing target QPS.
```

Comment:
Add more detailed explanation describing the algorithm, eg if/when threads wait etc.